imports/wasi_snapshot_preview1/fs.go

```diff
@@ -870,9 +870,7 @@ func fdReaddirFn(_ context.Context, mod api.Module, params []uint64) syscall.Err
 
 	// Add entries for dot and dot-dot as wasi-testsuite requires them.
 	if cookie == 0 && dirents == nil {
-		if f, ok := fsc.LookupFile(fd); !ok {
```

Comment:
this is unnecessary because we already set `f` from the corresponding `fd` earlier:

```
	f, errno := openedDir(fsc, fd)
```